### DISCUSSION TITLE
Migration to Sisu

### DISCUSSION THREAD

**Discussion by slawekjaranowski** [codehaus-plexus/plexus-resources#5] [General]
Plexus IoC is now obsolete and looks like deprecated - what do you think about migrate plexus-resources to Sisu?

**Comment by rfscholte** [comment]
I wondered what still depends on `plexus-container-default`, so I removed it.
Issues are mostly related to logging (which can be replaced to slf4j-api).
The other is related to an 'InitializationException', which most likely can be replaced too.
So I'd say don't migrate to sisu but drop `plexus-container-default`.

**Reply by slachiewicz** [reply]
some time ago, on maven-dev we had a discussion that we shouldn't use slf4j

**Reply by rfscholte** [reply]
IIRC that discussion was about plugins. For Maven plugins there's a separate logging API.
For the Plexus projects SLF4J is most likely the best option.

**Reply by slachiewicz** [reply]
Yes, that was that case - but plugins uses our plexus components so indirectly slf4j would be prerequisite. That's my side comment only.

**Comment by slawekjaranowski** [comment]
There is another class from `plexus-container-default` - `org.codehaus.plexus.PlexusTestCase`.

It will be good to have possibility to replace it by something like: 

https://github.com/apache/maven/blob/master/maven-test-support/src/main/java/org/apache/maven/test/PlexusExtension.java

Do you thought to release https://github.com/apache/maven/tree/master/maven-test-support as separate project ... or may be this should be somewhere in sisu ...

**Reply by rfscholte** [reply]
Looking at that code it totally makes sense to move those classes to codehaus-plexus or sisu. Will need to discuss that with gnodet.

**Reply by gnodet** [reply]
@slawekjaranowski @rfscholte  Agreed, the fact that those classes are located in maven is mainly because sending PRs involving multiple repositories is quite a pain, so the easiest way for me to provide these changes.

**Reply by rfscholte** [reply]
@gnodet what do you prefer? In case of codehaus-plexus I can easily create a new repository, e.g. `plexus-test` or `plexus-testing` and let you add those classes. Pushing a first release can be done quite fast.

**Reply by gnodet** [reply]
I guess a new `plexus-testing` project would be a good start.  We can do a first release in order to be able to release maven 4.0.0-M1 soon.  I suppose a better plan would be to move it to sisu along with the core implementation though.

**Reply by rfscholte** [reply]
I agree. I've created https://github.com/codehaus-plexus/plexus-testing

**Reply by slachiewicz** [reply]
That will be repo i fact, for testing Jsr330 and not Plexus as we should soon migrate everything off Plexus components

**Reply by gnodet** [reply]
@rfscholte the repo is empty and I can not create a PR or push anything afaik. Can you just create an empty repo or add a simple readme ?

**Reply by gnodet** [reply]
I've raised https://github.com/codehaus-plexus/plexus-testing/pull/1

**Comment by slachiewicz** [comment]
Migration is done in https://github.com/codehaus-plexus/plexus-resources/commit/e129f80b15c4935c705a9fbdcc8d7ec0f38215a5 and released as version 1.2
